### PR TITLE
For #120: get test bundle

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #120 I've created classes to download and unpack test bundles. I've made this classes separate to make it possible do not download `.tar.gz` each time while working on local machine.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#146](https://github.com/artipie/rpm-adapter/issues/146) is assigned to @hdouss[/z](https://www.0crat.com/u/hdouss), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/a4d71c96-01ee-4576-9a66-4d1503d1c792); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/rpm/files/DownloadBundle.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            Channels.newChannel(URI.create(this.type.url).toURL().openStream()),
+            0, Long.MAX_VALUE
+        );
+        return gzip;
+    }
+
+    /**
+     * Bundle size.
+     */
+    enum Size {
```
Comment (HDouss) [inline src/main/java/com/artipie/rpm/files/DownloadBundle.java:69]: @olenagerasimova I am not getting this enum utility. Is DownloadBundle intended to download only these url? Isn't better to have an `URL` as attribute instead of `Size`. Especially if the issue description is talking about http://mirror.centos.org/centos/7/os/x86_64/Pac...

File: src/main/java/com/artipie/rpm/files/DownloadBundle.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    public DownloadBundle(final Size type) {
+        this.type = type;
+    }
+
+    @Override
+    public Path value() throws IOException {
+        final Path gzip = Files.createTempDirectory("downloads").resolve("bundle.tar.gz");
+        gzip.toFile().createNewFile();
+        FileChannel.open(gzip).transferFrom(
+            Channels.newChannel(URI.create(this.type.url).toURL().openStream()),
```
Comment (HDouss) [inline src/main/java/com/artipie/rpm/files/DownloadBundle.java:60]: @olenagerasimova We could have used TeeInput, TeeInputStream or TeeOutput here

File: src/main/java/com/artipie/rpm/files/UnpackGzip.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        final GzipCompressorInputStream input =
+            new GzipCompressorInputStream(Files.newInputStream(this.gzip));
+        try (TarArchiveInputStream tar = new TarArchiveInputStream(input)) {
+            TarArchiveEntry entry;
+            while ((entry = (TarArchiveEntry) tar.getNextEntry()) != null) {
+                final File next = res.resolve(entry.getName()).toFile();
+                if (entry.isDirectory()) {
+                    next.mkdirs();
+                } else {
+                    try (OutputStream out = Files.newOutputStream(next.toPath())) {
```
Comment (HDouss) [inline src/main/java/com/artipie/rpm/files/UnpackGzip.java:70]: @olenagerasimova Also here we are using bad practices non-OOP public static classes. In cactoos, we have GzipInput, GzipOutput and Tee classes that could be used.

**Comment by olenagerasimova** [comment]
@HDouss thanks, corrected, please, take a look

**Comment by HDouss** [comment]
@g4s8 Good to merge

File: src/main/java/com/artipie/rpm/files/DownloadBundle.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+ */
+package com.artipie.rpm.files;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.cactoos.Scalar;
```
Comment (g4s8) [inline src/main/java/com/artipie/rpm/files/DownloadBundle.java:32]: @olenagerasimova please don't use Cactoos in main `src/main` code, since it doesn't have stable API interfaces, we have faced few critical bugs because of Cactoos dependencies conflicts (it was `SolidScalar` vs `Solid` as I remember). 

Update: this comment seems to be irrelevant because this class can be located at `src/test`

File: src/main/java/com/artipie/rpm/files/DownloadBundle.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+import org.cactoos.io.OutputStreamTo;
+import org.cactoos.io.TeeInputStream;
+
+/**
+ * Downloads test bundle into temp dir.
+ * @since 0.8
+ * @todo #120:30min Create test for this class and use it in RpmITCase to download and
+ *  UnpackGzip to unpack rmps from tar.gz. Use longTests profile for mentioned tests.
+ */
+public final class DownloadBundle implements Scalar<Path> {
```
Comment (g4s8) [inline src/main/java/com/artipie/rpm/files/DownloadBundle.java:42]: @olenagerasimova why this class was created in main directory, not test? I think this class is suitable only for integration testing.

File: src/main/java/com/artipie/rpm/files/UnpackGzip.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.compress.utils.IOUtils;
+import org.cactoos.Scalar;
+import org.cactoos.io.OutputStreamTo;
+
+/**
+ * Unpacks given gzip to the temp dir.
+ * @since 0.8
+ */
+public final class UnpackGzip implements Scalar<Path> {
```
Comment (g4s8) [inline src/main/java/com/artipie/rpm/files/UnpackGzip.java:42]: @olenagerasimova same here - why `src/main`?

Comment (olenagerasimova) [reply src/main/java/com/artipie/rpm/files/UnpackGzip.java:42]: @g4s8 I think we will need this class to unpack metadata files from repository while updating it. I can simply not implement `Scalar` and get rid of cactoos usages

Comment (g4s8) [reply src/main/java/com/artipie/rpm/files/UnpackGzip.java:42]: @olenagerasimova ok, let's keep it here then

**Comment by olenagerasimova** [comment]
@g4s8 thanks, please, take a look: I've moved `Download` to test package, renamed `UnpackGzip` and got rid of cactoos usages there.

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @hdouss[/z](https://www.0crat.com/u/hdouss), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good
